### PR TITLE
Gsc multiple properties

### PR DESCRIPTION
**Pull request by enjoyear**
The change is to support fetching multiple google search console properties in one gobblin job. Also include the source property as the last column of the schema.

### CODE REVIEW

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java [hunk 1]
```diff
@@ -47,7 +50,12 @@
    * Must Provide.
    * Provide the property site URL whose google search analytics data you want to download
    */
-  public static final String KEY_PROPERTY = SOURCE_GOOGLE_WEBMASTER_PREFIX + "property_url";
+  public static final String KEY_PROPERTY = SOURCE_GOOGLE_WEBMASTER_PREFIX + "property_urls";
+  /**
+   * Optional. Default to true.
```
Comment (abti) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: I think typical use-case would be to honor source schema, so any additional columns added at runtime should not be a default behavior

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java [hunk 2]
```diff
@@ -159,14 +167,17 @@
   // =============================================
 
   private final static Splitter splitter = Splitter.on(",").omitEmptyStrings().trimResults();
+  private final static String SOURCE_PROPERTY_COLUMN_NAME = "source";
```
Comment (abti) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: What if source schema has "source" column name? Should we make this configurable?

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java [hunk 3]
```diff
@@ -47,7 +50,12 @@
    * Must Provide.
    * Provide the property site URL whose google search analytics data you want to download
    */
-  public static final String KEY_PROPERTY = SOURCE_GOOGLE_WEBMASTER_PREFIX + "property_url";
+  public static final String KEY_PROPERTY = SOURCE_GOOGLE_WEBMASTER_PREFIX + "property_urls";
+  /**
+   * Optional. Default to true.
+   * Determine whether to add source property as the last column to your configured schema
+   */
+  public static final String KEY_INCLUDE_SOURCE_PROPERTY = SOURCE_GOOGLE_WEBMASTER_PREFIX + "include_source_property";
```
Comment (abti) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: Do you also intend to include more columns to schema at runtime in future? If so, its better to move to a model where additional columns are added to schema via converter. If not, we can stick to what you have.

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSourceDaily.java [hunk 4]
```diff
@@ -45,7 +46,7 @@
   @Override
   GoogleWebmasterExtractor createExtractor(WorkUnitState state, Map<String, Integer> columnPositionMap,
       List<GoogleWebmasterFilter.Dimension> requestedDimensions,
-      List<GoogleWebmasterDataFetcher.Metric> requestedMetrics)
+      List<GoogleWebmasterDataFetcher.Metric> requestedMetrics, JsonArray schemaJson)
```
Comment (abti) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSourceDaily.java]: I see that you have changes some contracts (constructors), are the interfaces / classes in these connectors still evolving. If so, we should mark the classes here with @ Alpha annotation

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java (hunk 1 above)
Comment (enjoyear) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: Okay. I will change default behavior to false.

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java (hunk 3 above)
Comment (enjoyear) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: No, I think that's the only usecase I can think of now.

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java (hunk 2 above)
Comment (enjoyear) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSource.java]: I will make it configurable.

File: gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSourceDaily.java (hunk 4 above)
Comment (enjoyear) [inline gobblin-modules/google-ingestion/src/main/java/gobblin/ingestion/google/webmaster/GoogleWebMasterSourceDaily.java]: Will do.

**Comment by enjoyear** [comment]
@abti , ready for another review. Thanks!

**Pull request merged**